src/types/iterator.rs

```diff
@@ -107,65 +107,62 @@ mod tests {
 
     #[test]
     fn vec_iter() {
-        let gil_guard = Python::acquire_gil();
-        let py = gil_guard.python();
-        let obj = vec![10, 20].to_object(py);
-        let inst = obj.as_ref(py);
-        let mut it = inst.iter().unwrap();
-        assert_eq!(10, it.next().unwrap().unwrap().extract().unwrap());
-        assert_eq!(20, it.next().unwrap().unwrap().extract().unwrap());
-        assert!(it.next().is_none());
+        Python::with_gil(|py| {
+            let obj = vec![10, 20].to_object(py);
+            let inst = obj.as_ref(py);
+            let mut it = inst.iter().unwrap();
+            assert_eq!(10, it.next().unwrap().unwrap().extract().unwrap());
+            assert_eq!(20, it.next().unwrap().unwrap().extract().unwrap());
+            assert!(it.next().is_none());
+        });
     }
 
     #[test]
     fn iter_refcnt() {
-        let obj;
-        let count;
-        {
-            let gil_guard = Python::acquire_gil();
-            let py = gil_guard.python();
-            obj = vec![10, 20].to_object(py);
-            count = obj.get_refcnt(py);
-        }
+        let (obj, count) = Python::with_gil(|py| -> (Py<PyAny>, isize) {
```

Comment:
Looks like you'll either need to tweak imports for abi3 or remove the closure return type.

```suggestion
        let (obj, count) = Python::with_gil(|py| {
```

Reply:
Right that probably should have been a c_int anyway